bindings/java/src/org/sleuthkit/datamodel/WindowsAccountUtils.java

```diff
@@ -171,26 +266,173 @@ static boolean isWindowsSpecialSid(String sid) {
+	 * 
+	 * @param sid SID to check.
+	 * @return Realm Name for Windows special SID, an empty string if the SID is not a known special SID. 
+	 * 
+	 * @throws TskCoreException 
 	 */
-	static String getWindowsSpecialSidName(String sid) {
+	private static String getWindowsWellKnownSidRealmAddr(String sid) throws TskCoreException {
+		String tempSID = stripWindowsBackupPostfix(sid);
+
+		if (SPECIAL_SIDS_MAP.containsKey(tempSID)) {
+			return SPECIAL_SIDS_MAP.get(tempSID).getRealmAddr();
+		}
+		
+		for (Entry<String, WellKnownSidInfo> specialPrefixEntry : SPECIAL_SID_PREFIXES_MAP.entrySet()) {
+			if (tempSID.startsWith(specialPrefixEntry.getKey())) {
+				return specialPrefixEntry.getValue().getRealmAddr();
+			}
+		}
+
+		Matcher match = WINDOWS_SPECIAL_ACCOUNT_PREFIX_REGEX.matcher(tempSID);
+		if (match.find()) {
+			Integer domainIdentifier = Integer.valueOf(match.group(1));
+			// All the prefixes in the range S-1-5-80 to S-1-5-111 are special
+			if (domainIdentifier != null && domainIdentifier >= 80 && domainIdentifier <= 111) {
+				String realmAddr = String.format("%s-%d", NTAUTHORITY_SID_PREFIX, domainIdentifier);
+				return realmAddr;
+			}
+		}
+		
+		return "";
+	}
+	/**
+	 * Get the well known SID info for the given SID. 
+	 * 
+	 * @param sid SID to check.
+	 * 
+	 * @return WellKnownSidInfo for the SID, null if there is no info available. 
+	 */
+	private static WellKnownSidInfo getWindowsWellKnownInfo(String sid) {
 		String tempSID = stripWindowsBackupPostfix(sid);
 		
 		if (SPECIAL_SIDS_MAP.containsKey(tempSID)) {
 			return SPECIAL_SIDS_MAP.get(tempSID);
 		}
-		for (Entry<String, String> specialPrefixEntry: SPECIAL_SID_PREFIXES_MAP.entrySet()) {
+		for (Entry<String, WellKnownSidInfo> specialPrefixEntry: SPECIAL_SID_PREFIXES_MAP.entrySet()) {
 			if (tempSID.startsWith(specialPrefixEntry.getKey())) {
 				return specialPrefixEntry.getValue();
 			}
 		}
-		return "";
+		return null;
+	}
+	
+	/**
+	 * Get the realm address for the given special Windows SID.
+	 * 
+	 * @param sid SID to check.
+	 * 
+	 * @return Name for Windows special SID, an empty string if the SID is not a known special SID. 
+	 */
+	static String getWindowsWellKnownSidFullName(String sid) {
+		WellKnownSidInfo wellKnownSidInfo = getWindowsWellKnownInfo(sid);
+		return Objects.nonNull(wellKnownSidInfo) ? wellKnownSidInfo.getDescription() : "";
+	}
+	
+	/**
+	 * Get the realm name for the given well known Windows SID.
+	 * 
+	 * @param sid SID to check.
+	 * 
+	 * @return Realm Name for Windows special SID, an empty string if the SID is not a known special SID. 
+	 */
+	static String getWindowsWellKnownSidRealmName(String sid) {
+		
+		if (StringUtils.isNotBlank(sid) && sid.equals(NTAUTHORITY_SID_PREFIX)) {
+			return NTAUTHORITY_REALM_NAME;
+		}
+		
+		WellKnownSidInfo wellKnownSidInfo = getWindowsWellKnownInfo(sid);
+		return Objects.nonNull(wellKnownSidInfo) 
+				? wellKnownSidInfo.getRealmName() 
+				: null;
```

Comment:
should this be returning empty string per the functions comments?

Reply:
fixed comment.